torch/onnx/utils.py

```diff
@@ -172,6 +172,20 @@ def setup_onnx_logging(verbose: bool):
             torch.onnx.disable_log()
 
 
+@contextlib.contextmanager
+@_beartype.beartype
+def _setup_onnx_diagnostic():
+    engine = diagnostics.engine
+    _previous_context = diagnostics.context
+    diagnostics.context = engine.start_diagnostic_context(
+        diagnostics.ExportDiagnosticTool()
+    )
+    try:
+        yield diagnostics.context
+    finally:
+        diagnostics.context = _previous_context
```

Comment:
What is the desired behavior for managing different contexts? When do we want to resume the previous context?

Reply:
Modified the code a bit wrt to variable names. I added docstring in `ExportDiagnosticEngine` for explanation. This function will likely move to the same file as `ExportDiagnosticEngine` per our other discussion.